x/budget/keeper/budget.go

```diff
@@ -10,7 +10,11 @@ import (
 // CollectBudgets collects all the valid budgets registered in params.Budgets and
 // distributes the total collected coins to collection address.
 func (k Keeper) CollectBudgets(ctx sdk.Context) error {
-	budgets := k.CollectibleBudgets(ctx)
+	params := k.GetParams(ctx)
+	var budgets []types.Budget
+	if params.EpochBlocks > 0 && ctx.BlockHeight()%int64(params.EpochBlocks) == 0 {
+		budgets = k.CollectibleBudgets(ctx, params.Budgets)
+	}
```

Comment:
Could you explain why this function and `CollectibleBudgets` are modifed? At first glance, it seems the logic is not changed but the structure has been changed.

Reply:
Originally, while `CollectibleBudgets` was being refactored to recycle `CollectibleBudgets` from `ValidateBudgets`, but it was stopped when I decided to create the overlap function separately. 
so `CollectibleBudgets` could be reverted or continue to refactor to as a type function without `ctx` dependency, What do you think?

Reply:
Both make sense, but I'd recommend to revert back for now.
Maybe later we can refactor `CollectibleBudgets` to not rely on `sdk.Context` so that we can test that function more easily.

Reply:
Sure, I just committed the addtional refactoring code